tests/test_models.py

```diff
@@ -407,3 +407,48 @@ def test_time_intervals_invalid(interval) -> None:
 def test_time_intervals_valid(interval) -> None:
     """Check Time Interval model."""
     assert TimeInterval(interval=interval)
+
+
+@pytest.mark.parametrize(
+    "bboxes",
+    [
+        # invalid Y order
+        [[0, 1, 1, 0]],
+        # sub-sequent crossing Y
+        [[0, 0, 2, 2], [0.5, 0.5, 2.0, 2.5]],
+        # sub-sequent crossing X
+        [[0, 0, 2, 2], [0.5, 0.5, 2.5, 2.0]],
+        # sub-sequent crossing Antimeridian limit
+        [[0, 0, 2, 2], [1, 0, -179, 1]],
+        # both crossing Antimeridian limit but sub-sequent cross has min lat -176 > -178
+        [[2, 0, -178, 2], [1, 0, -176, 1]],
+        # sub-sequent cross Antimeridian but not the overall
+        [[0, 0, 2, 2], [1, 0, -176, 1]],
+    ],
+)
+def test_spatial_intervals_invalid(bboxes) -> None:
+    """Check invalid Spatial Interval model."""
+    with pytest.raises(ValidationError):
+        SpatialExtent(bbox=bboxes)
+
+
+@pytest.mark.parametrize(
+    "bboxes",
+    [
+        [[0, 0, 1, 1]],
+        # Same on both side
+        [[0, 0, 2, 2], [0, 0, 2, 2]],
+        [[0, 0, 2, 2], [0.5, 0.5, 1.5, 1.5]],
+        # crossing Antimeridian limit
+        [[2, 0, -178, 2]],
+        # overall crossing Antimeridian, sub-sequent bbox not crossing (but within overall right part)
+        [[2, 0, -178, 2], [0, 0, 1, 1]],
```

Comment:
The antimeridian crossing has -180/180 as the centerline, so the interior needs to be really close to that.

```suggestion
        [[2, 0, -178, 2], [-179, -178, 1, 1]],
```